_drafts/_draft-2024-02-22-connectivity-metrics.md

```diff
@@ -0,0 +1,232 @@
+
+### Logging
+
+Logging is a foundational tool for understanding the behavior of embedded devices’ connectivity. By capturing key events and status updates, logs summarize what transpires within the connectivity stack. The following example is from an ESP32S3 connecting to my home wifi network:
+
+```text
+I (454) mflt: Data poster task up and running every 60s.
+I (494) pp: pp rom version: e7ae62f
+I (504) net80211: net80211 rom version: e7ae62f
+I (514) wifi:wifi driver task: 3fcb4c08, prio:23, stack:6656, core=0
+I (524) wifi:wifi firmware version: 9aad399
+I (524) wifi:wifi certification version: v7.0
+I (524) wifi:config NVS flash: enabled
+I (524) wifi:config nano formating: disabled
+I (524) wifi:Init data frame dynamic rx buffer num: 32
+I (544) wifi:Init management frame dynamic rx buffer num: 32
+I (544) wifi:Init management short buffer num: 32
+I (554) wifi:Init dynamic tx buffer num: 32
+I (554) wifi:Init static tx FG buffer num: 2
+I (554) wifi:Init static rx buffer size: 1600
+I (554) wifi:Init static rx buffer num: 10
+I (564) wifi:Init dynamic rx buffer num: 32
+I (574) wifi_init: rx ba win: 6
+I (574) wifi_init: tcpip mbox: 32
+I (574) wifi_init: udp mbox: 6
+I (574) wifi_init: tcp mbox: 6
+I (584) wifi_init: tcp tx win: 5744
+I (594) wifi_init: tcp rx win: 5744
+I (594) wifi_init: tcp mss: 1440
+I (594) wifi_init: WiFi IRAM OP enabled
+I (604) wifi_init: WiFi RX IRAM OP enabled
+I (614) phy_init: phy_version 601,fe52df4,May 10 2023,17:26:54
+I (654) wifi:mode : sta (60:55:f9:f5:2a:d0)
+I (664) wifi:enable tsf
+I (674) wifi:new:<7,0>, old:<1,0>, ap:<255,255>, sta:<7,0>, prof:1
+I (674) wifi:state: init -> auth (b0)
+esp32> I (5024) wifi:state: auth -> init (200)
+I (5024) wifi:new:<7,0>, old:<7,0>, ap:<255,255>, sta:<7,0>, prof:1
+I (7474) wifi:new:<7,0>, old:<7,0>, ap:<255,255>, sta:<7,0>, prof:1
+I (7474) wifi:state: init -> auth (b0)
+I (8134) wifi:state: auth -> assoc (0)
+I (8134) wifi:state: assoc -> run (10)
+I (8154) wifi:connected with <my_network>, aid = 9, channel 7, BW20, bssid = <my_routers_mac_addr>
+I (8154) wifi:security: WPA3-SAE, phy: bgn, rssi: -49
+I (8164) wifi:pm start, type: 1
+
+I (8164) wifi:set rx beacon pti, rx_bcn_pti: 0, bcn_timeout: 0, mt_pti: 25000, mt_time: 10000
+I (8184) wifi:<ba-add>idx:0 (ifx:0, <my_routers_mac_addr>), tid:6, ssn:0, winSize:64
+I (8184) wifi:AP's beacon interval = 102400 us, DTIM period = 1
+I (8904) wifi:<ba-add>idx:1 (ifx:0, <my_routers_mac_addr>), tid:5, ssn:2, winSize:64
+I (9174) esp_netif_handlers: sta ip: 192.168.50.235, mask: 255.255.255.0, gw: 192.168.50.1
+I (9174) wifi station: got ip:192.168.50.235
+I (9174) wifi station: connected to ap SSID:<my_network>
+I (9874) mflt: Result: 0
+I (9874) mflt: Checking for OTA Update
+I (10494) mflt: Up to date!
+```
```

Comment:
Need to add description of what these logs are useful for

Reply:
You could break it up into a couple segments and give some brief explanation. Here's a template:

First we see some information about the wifi chip, the driver task, and config data. This could be useful if we aren't already collection these version and configuration values, and need to know ...

```
I (454) mflt: Data poster task up and running every 60s.
I (494) pp: pp rom version: e7ae62f
I (504) net80211: net80211 rom version: e7ae62f
I (514) wifi:wifi driver task: 3fcb4c08, prio:23, stack:6656, core=0
I (524) wifi:wifi firmware version: 9aad399
I (524) wifi:wifi certification version: v7.0
I (524) wifi:config NVS flash: enabled
I (524) wifi:config nano formating: disabled
```

Next we see some information on the data storage for the buffers, which might help us if ...

```
I (524) wifi:Init data frame dynamic rx buffer num: 32
I (544) wifi:Init management frame dynamic rx buffer num: 32
I (544) wifi:Init management short buffer num: 32
I (554) wifi:Init dynamic tx buffer num: 32
I (554) wifi:Init static tx FG buffer num: 2
I (554) wifi:Init static rx buffer size: 1600
I (554) wifi:Init static rx buffer num: 10
I (564) wifi:Init dynamic rx buffer num: 32
I (574) wifi_init: rx ba win: 6
```

And then finally we jump into some logs showing the chip trying to connect to an access point. Here is where we might see a few clues about failed authentication, but this connection was successful. We see at the end we were assigned an IP address ✨ .

```
I (674) wifi:new:<7,0>, old:<1,0>, ap:<255,255>, sta:<7,0>, prof:1
I (674) wifi:state: init -> auth (b0)
esp32> I (5024) wifi:state: auth -> init (200)
I (5024) wifi:new:<7,0>, old:<7,0>, ap:<255,255>, sta:<7,0>, prof:1
I (7474) wifi:new:<7,0>, old:<7,0>, ap:<255,255>, sta:<7,0>, prof:1
I (7474) wifi:state: init -> auth (b0)
I (8134) wifi:state: auth -> assoc (0)
I (8134) wifi:state: assoc -> run (10)
I (8154) wifi:connected with <my_network>, aid = 9, channel 7, BW20, bssid = <my_routers_mac_addr>
I (8154) wifi:security: WPA3-SAE, phy: bgn, rssi: -49
I (8164) wifi:pm start, type: 1

I (8164) wifi:set rx beacon pti, rx_bcn_pti: 0, bcn_timeout: 0, mt_pti: 25000, mt_time: 10000
I (8184) wifi:<ba-add>idx:0 (ifx:0, <my_routers_mac_addr>), tid:6, ssn:0, winSize:64
I (8184) wifi:AP's beacon interval = 102400 us, DTIM period = 1
I (8904) wifi:<ba-add>idx:1 (ifx:0, <my_routers_mac_addr>), tid:5, ssn:2, winSize:64
I (9174) esp_netif_handlers: sta ip: 192.168.50.235, mask: 255.255.255.0, gw: 192.168.50.1
I (9174) wifi station: got ip:192.168.50.235
I (9174) wifi station: connected to ap SSID:<my_network>
```